ably-ios/ARTRest.m

```diff
@@ -200,10 +207,20 @@ - (void)time:(void(^)(NSDate *time, NSError *error))callback {
     return nil;
 }
 
-- (void)stats:(ARTStatsQuery *)query callback:(void (^)(ARTPaginatedResult *, NSError *))callback {
-    NSParameterAssert(query.limit < 1000);
-    NSParameterAssert([query.start compare:query.end] != NSOrderedDescending);
-    
+- (BOOL)stats:(ARTStatsQuery *)query callback:(void (^)(ARTPaginatedResult *, NSError *))callback error:(NSError **)errorPtr {
+    if (query.limit > 1000 && errorPtr) {
```

Comment:
I think that's pre-ARC docs. I am not on my laptop to check that with the
compiler but sure I will do it later.

On Thursday, 24 December 2015, Toni Cárdenas <notifications@github.com>
wrote:

> In ably-ios/ARTRest.m
> <https://github.com/ably/ably-ios/pull/85#discussion_r48424487>:
>
> > @@ -200,10 +207,20 @@ - (void)time:(void(^)(NSDate *time, NSError *error))callback {
> >      return nil;
> >  }
> >
> > -- (void)stats:(ARTStatsQuery *)query callback:(void (^)(ARTPaginatedResult *, NSError *))callback {
> > -    NSParameterAssert(query.limit < 1000);
> > -    NSParameterAssert([query.start compare:query.end] != NSOrderedDescending);
> > -
> > +- (BOOL)stats:(ARTStatsQuery *)query callback:(void (^)(ARTPaginatedResult *, NSError *))callback error:(NSError **)errorPtr {
> > +    if (query.limit > 1000 && errorPtr) {
>
> I get that if errorPtr is nil then the caller ignores the error, but I
> don't think that we should just continue when the preconditions are wrong.
> We should return early to avoid an inconsistent state that could even crash
> the program.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ably/ably-ios/pull/85/files#r48424487>.
>